packages/react-form/src/hooks/list/baselist.ts

```diff
@@ -86,5 +94,20 @@ export function useBaseList<Item extends object>(
     });
   }, [state.list, handlers]);
 
-  return {fields, dispatch};
+  const listWithoutFieldStates: Item[] = useMemo(() => {
+    return state.list.map(item => {
+      return mapObject(item, field => field.value);
+    });
+  }, [state.list]);
+
+  const isBaseListDirty = useMemo(
+    () => !isEqual(listWithoutFieldStates, state.initial),
+    [listWithoutFieldStates, state.initial],
+  );
+
+  const fieldsDirty = useDirty({fields});
+
+  const dirty = fieldsDirty || isBaseListDirty;
```

Comment:
nit: I wonder if some of these simpler values might actually be nicer just as part of the object literal we return, so there's less to hold in your head / scroll up and reference when you're trying to follow the code